libs/libc/tls/tls_getset.c

```diff
@@ -56,16 +53,14 @@
 
 tls_ndxset_t tls_get_set(void)
 {
-  FAR struct tcb_s *rtcb = this_task();
-  FAR struct task_group_s *group = rtcb->group;
-  irqstate_t flags;
+  FAR struct task_info_s *tinfo = task_get_info();
   tls_ndxset_t tlsset;
 
-  DEBUGASSERT(group != NULL);
+  DEBUGASSERT(tinfo != NULL);
 
-  flags = spin_lock_irqsave(NULL);
-  tlsset = group->tg_tlsset;
-  spin_unlock_irqrestore(NULL, flags);
+  sem_wait(&tinfo->ta_tlssem);
+  tlsset = tinfo->ta_tlsset;
```

Comment:
What if the semaphore wait is awakened by a receipt of a signal?  Or if the thread is canceled while waiting?